From cc214e676ec172cd2ff6b8cd7165e462ae5ea7ad Mon Sep 17 00:00:00 2001
From: Dan Pasanen <dan.pasanen@gmail.com>
Date: Wed, 23 Aug 2017 08:32:09 -0500
Subject: [PATCH 6/7] soong: use absolute path for OUT_DIR

* This is needed for tasks like inline kernel building. If
  OUT_DIR_COMMON_BASE (which will be an absolute path) is not
  set, prepend TOP to "out" in order to always use a full path.

Change-Id: I5f955c42854922425268f4cd54dec8cf0a531195
Signed-off-by: SagarMakhar <sagarmakhar@gmail.com>
Signed-off-by: Kunmun <kunmun.devroms@gmail.com>
---
 ui/build/config.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ui/build/config.go b/ui/build/config.go
index 862e09f3c..dadc649a3 100644
--- a/ui/build/config.go
+++ b/ui/build/config.go
@@ -146,6 +146,8 @@ func NewConfig(ctx Context, args ...string) Config {
 			} else {
 				outDir = filepath.Join(baseDir, filepath.Base(wd))
 			}
+		} else {
+			outDir = filepath.Join(os.Getenv("TOP"), outDir)
 		}
 		ret.environ.Set("OUT_DIR", outDir)
 	}
-- 
2.17.1

